Upgrade wp-prettier to v3.0.3

jsnajdr opened the pull request:
Upgrades the `wp-prettier` fork to version 3.0.3.

Most visible formatting changes are:
- fixed bug where parenthesized and broken `( await x )` expression had an extra trailing space
- fixed bug where function params with only a comment (`call( /* none */ )`) didn't have paren spaces
- fixed bug where union types (`( string | undefined )`) had an extra trailing space
- many function calls now have a trailing comma in argument list where it was previously missing
- assignments (`const x = y;`) are now formatted more nicely, often breaking after the `=`

In the `.prettierrc` config file, I had to add an explicit `trailingComma: 'es5'` config because the default has changed to `all` in Prettier 3.0, and that would lead to added commas in more than 4000 files. Gutenberg also has this config option set explicitly.

The PR also upgrades `eslint-plugin-prettier` to version 5, because the previous versions were not compatible with breaking changes in Prettier 3.0's API.

**How to test:**
Go through the list of the formatting changes and report anything suspicious.

jsnajdr committed:
Upgrade wp-prettier to 3.0.3 beta 4

jsnajdr committed:
Add explicing trailingComma: es5 config

jsnajdr committed:
Upgrade eslint-plugin-prettier to 5.0.0

jsnajdr commented on packages/domain-picker/src/components/index.tsx:
```diff
@@ -209,7 +209,7 @@ const DomainPicker: FunctionComponent< Props > = ( {
 	}, [ initialDomainSearch, showSearchField ] );
 
-	const suggestionRefs = React.useRef< ( HTMLButtonElement | null )[] >( [] );
+	const suggestionRefs = React.useRef< HTMLButtonElement | null[] >( [] );
```
Oops, this is a major bug that wasn't caught in Gutenberg. Adding the parens is critical here.

jsnajdr replied:
Fixed in `wp-prettier@3.0.3-beta-4`.

jsnajdr committed:
Reformat code with Prettier 3.0.3

tyxla committed:
dedupe

tyxla commented on .prettierrc:
```diff
@@ -6,3 +6,4 @@ bracketSpacing: true
 parenSpacing: true
 bracketSameLine: false
 semi: true
+trailingComma: 'es5'
```
Confirming this is necessary since the default value changed from `es5` to `all` since `v3`.

tyxla commented on apps/happy-blocks/block-library/universal-header-v2/view.js:
```diff
@@ -175,12 +175,11 @@ import './style.scss';
-			) {
-				element.blur();
-			} );
+			this.find( '[tabindex], button, input, object, select, textarea' ).each(
+				function ( _, element ) {
+					element.blur();
+				}
+			);
```
Those snippets end up way more readable 💯

tyxla commented on client/lib/analytics/ga.js:
```diff
@@ -76,31 +76,28 @@ export const gaRecordPageView = makeGoogleAnalyticsTrackingFunction( function re
-		value = Math.round( Number( String( value ) ) ); // GA requires an integer value.
-		// https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#eventValue
-	}
+export const gaRecordEvent = makeGoogleAnalyticsTrackingFunction(
```
Here's another obvious readability improvement as a consequence of the update 👍

tyxla reviewed (approved):
Went through the changes and IMHO things look great 👍 There are obvious examples of improvement as a consequence of the update. 

The only thing I pushed was a `dedupe` since some packages were unnecessarily using multiple versions as a consequence of the update.

Nice work @jsnajdr 🚀

jsnajdr committed:
More formatting changes after rebase

jsnajdr commented:
Thanks @tyxla for careful review. My next steps are:
- rebased on latest `trunk` and run CI again
- added formatting changes in one more file that was merged to `trunk` while this PR was open
- wait for CI to finish and the merge 🚀
